File: anypick-python/tests/conftest.py
"""Shared test helpers: build Model/BenchmarkScore lists from fixtures."""

from __future__ import annotations

import json
import os
from typing import Any

from anypick.model import BenchmarkScore, Model
from anypick.openrouter import _map_benchmark, _map_model

FIXTURES = os.path.join(os.path.dirname(__file__), "fixtures")


def load_fixture(name: str) -> Any:
    with open(os.path.join(FIXTURES, name), "r", encoding="utf-8") as fh:
        return json.load(fh)


def models_from_fixture() -> list[Model]:
    payload = load_fixture("models.json")
    return [_map_model(item) for item in payload.get("data", [])]


def scores_from_fixture(task_type: str | None = "coding") -> list[BenchmarkScore]:
    payload = load_fixture("benchmarks.json")
    return [_map_benchmark(item, task_type) for item in payload.get("data", [])]

File: anypick-ts/tests/conftest.ts
/** Shared test helpers: build Model/BenchmarkScore lists from fixtures. */
import { readFileSync } from "node:fs";
import { join, dirname } from "node:path";
import { fileURLToPath } from "node:url";

import { BenchmarkScore, Model } from "../src/index.js";
import { mapBenchmark, mapModel } from "../src/openrouter.js";

const __dirname = dirname(fileURLToPath(import.meta.url));
const FIXTURES = join(__dirname, "fixtures");

export function loadFixture(name: string): any {
  return JSON.parse(readFileSync(join(FIXTURES, name), "utf8"));
}

export function modelsFromFixture(): Model[] {
  const payload = loadFixture("models.json");
  return (payload.data as Record<string, unknown>[]).map(mapModel);
}

export function scoresFromFixture(taskType: string | null = "coding"): BenchmarkScore[] {
  const payload = loadFixture("benchmarks.json");
  return (payload.data as Record<string, unknown>[]).map((item) =>
    mapBenchmark(item, taskType),
  );
}
